fedimint-api/src/module/mod.rs

```diff
@@ -198,8 +199,22 @@ where
     }
 }
 
+/// Logic responsible for module's initialization, config generation and validation
+///
+/// Once the module config is ready, the module can be instantiated via `[Self::init]`.
 #[async_trait]
-pub trait FederationModuleConfigGen {
+pub trait ModuleInit {
+    /// Initialize the [`ServerModule`] instance from its config
+    async fn init(
+        &self,
+        cfg: ServerModuleConfig,
+        db: Database,
+        task_group: &mut TaskGroup,
+        decoders: &ModuleDecoderRegistry,
```

Comment:
I don't think modules should need decoders since they don't care about other modules' data structures. I looked into why `Wallet` takes a `decoders` arg and it seems entirely about a DB tx that reads BTC tx from the DB to broadcast. We could use `ModuleDecoderRegistry::default()` there.